crates/sui/tests/shell_tests.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use fs_extra::dir::CopyOptions;
+use insta_cmd::get_cargo_bin;
+use std::fs;
+use std::path::Path;
+use std::process::Command;
+use sui_config::SUI_CLIENT_CONFIG;
+use test_cluster::TestClusterBuilder;
+
+// [test_shell_snapshot] is run on every file matching [TEST_PATTERN] in [TEST_DIR].
+// Files in [TEST_NET_DIR] will be run with a [TestCluster] configured.
+//
+// These run the files as shell scripts and compares their output to the snapshots; use `cargo
+// insta test --review` to update the snapshots.
+
+const TEST_DIR: &str = "tests/shell_tests";
+const TEST_NET_DIR: &str = "tests/shell_tests/with_network";
+const TEST_PATTERN: &str = r"\.sh$";
+
+/// run the bash script at [path], comparing its output to the insta snapshot of the same name.
+/// The script is run in a temporary working directory that contains a copy of the parent directory
+/// of [path], with the `sui` binary on the path.
+///
+/// If [cluster] is provided, the config file for the cluster is passed as the `CONFIG` environment
+/// variable.
+#[tokio::main]
+async fn test_shell_snapshot(path: &Path) -> datatest_stable::Result<()> {
+    // set up test cluster
+    let cluster = if path.starts_with(TEST_NET_DIR) {
```

Comment:
should this be `contains`? Unsure of the structure of `path` here though, so maybe `starts_with` is right too here.